### PR TITLE
 [VEN-1571]: Fix openZeppelin swap router audit

### PR DESCRIPTION
**Pull request by defcon022**
## Description
This PR resolves the OpenZepplein audit and fixes that we accepted to fix.

Audit fixes:
- L-01. Missing Docstring
- L-02. Locked BNB in contract
- N-01. Misleading Docstirng
- N-02. Naming can be improved
- N-04 Confusing Use of ETH and BNB in Comments and Function Names
- N-03 Some Convenience Functions Are Missing

-- In N-03 we have added the following methods:
• `swapExactTokensForBNBAndSupply`
• `swapExactTokensForBNBAndSupplyAtSupportingFee`
• `swapTokensForExactBNBAndSupply`

But we cannot implement `swapTokensForExactBNBAndSupplyAtSupportingFee` as  `swapTokensForExactBNBAtSupportingFee` does not exist because to get the amount of the tokens(deflationary) to swap for exactBNB is not possible.

Resolves #
VEN-1571

### CODE REVIEW

File: contracts/Swap/RouterHelper.sol
Comment (chechu) [inline contracts/Swap/RouterHelper.sol:1]: We have to add docs in this file, as part of L-01

File: contracts/Swap/lib/TransferHelper.sol
Comment (chechu) [inline contracts/Swap/lib/TransferHelper.sol:1]: We have to add docs in this file, as part of L-01

Also in PancakeLibrary and RouterHelper

File: contracts/Swap/SwapRouter.sol [hunk 1]
```diff
@@ -863,4 +935,16 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
             revert VTokenNotListed(vToken);
         }
     }
+
+    /**
+     * @notice Mint vBNB tokens to the market then transfer them to user
+     * @param swapAmount Swapped BNB amount
+     */
+
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:943]:
```suggestion
```

File: contracts/Swap/SwapRouter.sol [hunk 2]
```diff
@@ -253,11 +253,83 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
...
+    /**
+     * @notice Swap tokens for Exact BNB and supply to a Venus market
+     * @param amountOut The amount of the tokens needs to be as output token.
+     * @param amountInMax The maximum amount of input tokens that can be taken for the transaction not to revert.
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @param deadline Unix timestamp after which the transaction will revert.
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
+     * @dev In case of swapping native BNB the first asset in path array should be the wBNB address
+     */
+
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:319]:
```suggestion
```

File: contracts/Swap/SwapRouter.sol [hunk 3]
```diff
@@ -253,11 +253,83 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
...
+    /**
+     * @notice Swap Exact deflationary tokens (a small amount of fee is deducted at the time of transfer of tokens) for BNB and supply to a Venus market
+     * @param amountIn The amount of tokens to swap.
+     * @param amountOutMin Minimum amount of tokens to receive.
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @param deadline Unix timestamp after which the transaction will revert.
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
+     * @dev In case of swapping native BNB the first asset in path array should be the wBNB address
+     */
+
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:293]:
```suggestion
```

File: contracts/Swap/SwapRouter.sol [hunk 4]
```diff
@@ -253,11 +253,83 @@ contract SwapRouter is Ownable2Step, RouterHelper, IPancakeSwapV2Router {
...
+    /**
+     * @notice Swap Exact tokens for BNB and supply to a Venus market
+     * @param amountIn The amount of tokens to swap.
+     * @param amountOutMin Minimum amount of tokens to receive.
+     * @param path Array with addresses of the underlying assets to be swapped
+     * @param deadline Unix timestamp after which the transaction will revert.
+     * @dev Addresses of underlying assets should be ordered that first asset is the token we are swapping and second asset is the token we receive
+     * @dev In case of swapping native BNB the first asset in path array should be the wBNB address
+     */
+
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:270]:
```suggestion
```

File: contracts/Swap/lib/TransferHelper.sol [hunk 5]
```diff
@@ -26,7 +26,7 @@ library TransferHelper {
         // bytes4(keccak256(bytes('transferFrom(address,address,uint256)')));
         (bool success, bytes memory data) = token.call(abi.encodeWithSelector(0x23b872dd, from, to, value));
         if (!(success && (data.length == 0 || abi.decode(data, (bool))))) {
-            revert TransferFromFailed();
+            revert SafeTransferFromFailed();
         }
     }
```
Comment (chechu) [inline contracts/Swap/lib/TransferHelper.sol:32]: Should we rename `safeTransferETH` to `safeTransferBNB`, in the context of N-04?

File: contracts/Swap/RouterHelper.sol
Comment (defcon022) [reply contracts/Swap/RouterHelper.sol:1]: Added, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/lib/TransferHelper.sol
Comment (defcon022) [reply contracts/Swap/lib/TransferHelper.sol:1]: Added, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/SwapRouter.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/SwapRouter.sol (hunk 2 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/SwapRouter.sol (hunk 3 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/SwapRouter.sol (hunk 4 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol]: Fixed, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

File: contracts/Swap/lib/TransferHelper.sol (hunk 5 above)
Comment (defcon022) [reply contracts/Swap/lib/TransferHelper.sol:53]: Yes, Fixed, 9751c85d7054485d5f082aa4ba48c7ce878d9a34

**Review by chechu — APPROVED** [review]

**Review by GitGuru7 — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Pull request merged**